tests/benchmark/compute/instruction/test_system.py

```diff
@@ -806,24 +811,37 @@ def test_selfdestruct_initcode(
         )
         + Op.PUSH1[1]
         + Op.ADD
-        + Op.JUMPI(len(code_prefix) - 1, Op.GT(iterations, Op.DUP1))
+        + Op.JUMPI(
+            len(code_prefix) - 1, Op.GT(Op.GAS, suffix_cost + loop_cost)
+        )
     )
     code = code_prefix + loop_body + code_suffix
 
     # The 0 storage slot is initialize to avoid creation costs in SSTORE above.
     code_addr = pre.deploy_contract(code=code, balance=100_000, storage={0: 1})
-    code_tx = Transaction(
-        to=code_addr,
-        gas_limit=gas_benchmark_value,
-        gas_price=10,
-        sender=pre.fund_eoa(),
-    )
 
-    post = {code_addr: Account(storage={0: 42})}  # Check for successful
+    max_iterations_per_tx = (tx_gas_limit - base_costs) // loop_cost
+    num_exec_txs = math.ceil(iterations / max_iterations_per_tx)
+
+    exec_txs = []
+    with TestPhaseManager.execution():
+        for _ in range(num_exec_txs):
+            exec_txs.append(
+                Transaction(
+                    to=code_addr,
+                    gas_limit=tx_gas_limit,
+                    sender=pre.fund_eoa(),
+                )
+            )
+
+    post = {
+        code_addr: Account(storage={0: len(exec_txs) + 1})
+    }  # Check for successful
     # execution.
-    state_test(
-        pre=pre,
+    benchmark_test(
         post=post,
-        tx=code_tx,
+        blocks=[
+            Block(txs=exec_txs),
+        ],
         expected_benchmark_gas_used=iterations * loop_cost + base_costs,
```

Comment:
This should be updated, since there are more than one transaction, so it would charge more than one `base_costs`.

the new total cost should be as follows:




```suggestion
        expected_benchmark_gas_used=(max_iterations_per_tx * loop_cost + base_costs) * num_exec_txs,
```

Reply:
Thank you! Changed it.

~Is there a way to detect this missing locally? When I fill, doesn't seems this to be checked. It is only checked in CI?~ I think this requries evmone.

Reply:
No, this should not required evmone, it should by default examine the gas usage as sanity check.